adding resources

mohsenfaraj opened the pull request:
"pain is certain, suffering is optional" -Sun Tzu
adding some more resources to have easier suffer experience in university as CE student :grin:

amirihusayn commented:
Thanks @mohsenfaraj. We'll review your changes as soon as possible. Hope your contribution continues.

mohsenfaraj commented:
@amirihusayn we will continue. we never forgot the effort you put on this project and i am still grateful for all of it.
feel free to DM me at telegram, i would appreciate to have a conversation with you.
best regards.

amirihusayn commented:
We are still working on [directory conventions](https://github.com/codegeeks-ir/courses/issues/1), But there is the basic one; Here it is:  

```text
[course]/ 
  | 
  └───resources/
  |        |
  |        └──[master]/
  |               |
  |               └──slides/
  |               |
  |               └──assignments/
  |               |     |
  |               |     └──[github]-[year]/
  |               |           |
  |               |           └──assignment-1.py
  |               |
  |               └──notes/
  |               |     |
  |               |     └──[subject]-[github]-[year].pdf
  |               |
  |               └──exams/
  |                    |
  |                    └──quiz/
  |                    |
  |                    └──mid/
  |                    |
  |                    └──final/
  └──README.md
```  

It's clear where the handwritten notes, slides or even exams should be. So Please:  
1- Apply these rules to your own changes 
2- Commit using `--amend` option and push them by `--force`. ( Because We avoid tracking history for binary files like `pdf` )

If You don't have enough time for these things, I do it personally; But considering that I am the only one who maintains [codegeeks-ir](https://github.com/codegeeks-ir/codegeeks-ir), this PR will be reviewed later.  

Thanks for your support @mohsenfaraj ; I'm really grateful for the resources you collected. If You'd like to help more, checkout the [issues](https://github.com/codegeeks-ir/codegeeks-ir/issues) and the [projects](https://github.com/orgs/codegeeks-ir/projects).

Have an epic day !

amirihusayn commented on project-management/Dr. Farid Ahmadi/IRM.Press.Project.Management.for.Modern.Information.Systems.eBook-LinG.pdf:
Consider that adding books for free public use, violates copy rights and It will makes problem for us. Please delete this file and commit your changes with `--amend` option to remove it from history also. @mohsenfaraj

mohsenfaraj replied:
@amirihusayn  Sorry my bad! i thought its a free book.
I removed the mentioned book and fixed the files with the directory conventions that you mentioned. except there is a problem that i have no idea who are the authors of these notes and what year they wrote it. sorry for that!

PS. I can't message you from telegram but a lot of things have changed and if you want we could talk about it. long story short i decided that this projects need to be prioritized and you have my support for this project.

Thanks for your patience and hard work. it means a lot for me.

mohsenfaraj commented:
@amirihusayn Sorry my bad! i thought its a free book.
I removed the mentioned book and fixed the files with the directory conventions that you mentioned. except there is a problem that i have no idea who are the authors of these notes and what year they wrote it. sorry for that!

PS. I can't message you from telegram but a lot of things have changed and if you want we could talk about it. long story short i decided that this projects need to be prioritized and you have my support for this project.

Thanks for your patience and hard work. it means a lot for me.

amirihusayn commented:
> except there is a problem that i have no idea who are the authors of these notes and what year they wrote it. sorry for that!  

File naming conventions for things like notes doesn't mean You are the author or when they are created. We agree these rules just for making file names meaningful and preventing bad names like: 
- `compiler-1.pdf`
- `compiler-2.pdf`
- `compiler-bla-bla.pdf` 
- `compiler-i-swear-its-original.pdf`

So feel free to name them like the convention. Thanks for your support @mohsenfaraj

mohsenfaraj commented:
All is set and good. ready for review!
while this is great move i feel like using github for storing pdfs is not a optimal solution as the files number and size grows over time. we could save files to other places (like a cloud system or google drive) and just link that here so there would be no need for pulling the whole repo just to add a pdf. i feel managing this way could be a bit easier. we could still simulate the naming conventions and folders by url and putting them in nested lists but there is still challenges like finding a good free, collaborative and reliable cloud provider. for now lets just stick to what we have and work on necessary things!
thanks! @amirihusayn

amirihusayn commented:
That's right. As I mentioned, Git is not good option to handle binary files like `pdf`. Some how this is cheapest solution considering priorities and development time; But our main goal is to reach a position where We make `markdown` open source edits of those notes and slides. 

So students will be able to read them every where they are like regular docs, instead of downloading those. This requires some more people to work on it. Thanks again @mohsenfaraj God bless your helps.
